src/state/achievements/helpers.ts

```diff
@@ -24,22 +25,22 @@ export const getUserPointIncreaseEvents = async (account: string) => {
 export const getAchievements = async (account: string): Promise<Achievement[]> => {
   const pointIncreaseEvents = await getUserPointIncreaseEvents(account)
   return pointIncreaseEvents.reduce((accum, event) => {
-    if (!campaignMap.has(event.args[2])) {
+    if (!campaignMap.has(event.args[2].toString())) {
       return accum
     }
 
-    const campaignMeta = campaignMap.get(event.args[2])
+    const campaignMeta = campaignMap.get(event.args[2].toString())
 
     return [
       ...accum,
       {
-        id: event.args[2],
+        id: event.args[2].toString(),
         type: campaignMeta.type,
         address: event.address,
         title: getAchievementTitle(campaignMeta),
         description: getAchievementDescription(campaignMeta),
         badge: campaignMeta.badge,
-        points: Number(event.args[1]),
+        points: Number(event.args[1].toString()),
```

Comment:
For the future please always deconstruct the variable. Using the index directly makes it virtually impossible without inspecting the return for other devs to understand what the variable is.

```
const [, points, campaignId] = event.args;
```

Reply:
Ok